javascript/app-event-analysis/src/components/Grid.tsx

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import React, { memo, useMemo } from 'react'
+
+import { useLineColors } from '../utils/useColors.js'
+import type { GridProps } from './Grid.types.js'
+import { GridLine } from './GridLine.js'
+import { GridLineType } from './GridLine.types.js'
+
+export const Grid: React.FC<GridProps> = memo(function Grid({
+	height,
+	width,
+	xScale,
+	yScale,
+}) {
+	const colors = useLineColors()
```

Comment:
We use the same thematic grid line color for all charts, so you can simplify the useLineColors object by removing the gridLine property and just getting it from thematic here in this component, now that this is a nice and self-contained thing.